WooCommerce/WooCommerceTests/ViewRelated/Search/Order/OrderSearchUICommandTests.swift

```diff
@@ -0,0 +1,171 @@
+        XCTAssertEqual(sanitizedKeywordWithoutHash, "123", "Expected sanitizeKeyword to return the keyword unchanged if there's no leading '#'")
+    }
+
+    func testSynchronizeModelsTracksAnalytics() throws {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager, analytics: analytics, stores: stores)
+        let keyword = "testKeyword"
+        stores.whenReceivingAction(ofType: OrderAction.self) { (action: OrderAction) in
+            guard case let .searchOrders(_, _, _, _, onCompletion) = action else {
+                return XCTFail("Unexpected action: \(action)")
+            }
+            onCompletion(nil)
+        }
+
+        // Mock order statuses
+        insertOrderStatuses()
+
+        // When
+        waitFor { promise in
+            command.synchronizeModels(siteID: self.siteID, keyword: keyword, pageNumber: 1, pageSize: 20) { success in
+                promise(())
+            }
+        }
+
+        // Then
+        guard let event = analyticsProvider.receivedEvents.first else {
+            XCTFail("Expected an event but found none")
+            return
+        }
+        XCTAssertEqual(event, "orders_list_search")
+
+        guard let eventProperties = analyticsProvider.receivedProperties.first else {
+            XCTFail("Expected event properties but found none")
+            return
+        }
+        XCTAssertEqual(eventProperties["search"] as? String, keyword)
+    }
+
+    func testDidSelectSearchResult() throws {
+        // Given
+        let order = Networking.Order.fake().copy()
+
+        // When
+        let selectedOrder = waitFor { promise in
+            let command = OrderSearchUICommand(siteID: self.siteID, onSelectSearchResult: { order, _ in
+                promise(order)
+            }, storageManager: storageManager)
+            command.didSelectSearchResult(model: order, from: .init(), reloadData: {}, updateActionButton: {})
+        }
+
+        // Then
+        XCTAssertEqual(selectedOrder, order)
+    }
+
+    private func insertOrderStatuses() {
+        let statuses: [OrderStatusEnum] = [.pending, .processing, .onHold, .completed, .cancelled, .failed, .custom("aCustomStatus")]
+        statuses.forEach { status in
+            storageManager.insertOrderStatus(name: status.rawValue)
+        }
+        storageManager.viewStorage.saveIfNeeded()
+    }
+
+    private func waitFor<T>(timeout: TimeInterval = 1, _ completion: (_ promise: @escaping (T) -> Void) -> Void) -> T? {
+        var result: T?
+        let expectation = self.expectation(description: "Waiting for completion")
+        completion {
+            result = $0
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: timeout, handler: nil)
+        return result
+    }
```

Comment:
Why do we need this? There's an [implementation in TestKit that we normally use](https://github.com/woocommerce/woocommerce-ios/blob/445a748833707ba989c465412b421b0a02948e7d/TestKit/Sources/TestKit/XCTestCase%2BWait.swift#L72) – I'm not sure whether you knew about that, or if this offers some different behaviour we want?

Anyway, if you choose to remove it you just need to fix some self-capture errors, and it seems to work correctly otherwise.

Reply:
I think I needed this because using the referenced implementation was producing errors.

Reply:
What errors do you get? I'm not seeing any. I've pushed up 2d6d131 with the changes to see what happens in CI... if it doesn't work, we can revert.

Reply:
Thanks for pushing that. Looks good and it's better to reuse the existing code 👍